src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -27,8 +27,8 @@ std::shared_ptr<ov::Node> change_constant_precision_to_fp16(std::shared_ptr<ov::
 
     int num_out_of_range = 0;
     for (size_t i = 0; i < size; ++i) {
-        // if it is smaller than the smallest positive normal fp16 but not zero
-        if (std::abs(src_data[i]) <= ov::float16::from_bits(0x0400) && src_data[i] != 0.0f) {
+        // if abs value is smaller than the smallest positive fp16, but not zero
+        if (std::abs(src_data[i]) < ov::float16::from_bits(0x0001) && src_data[i] != 0.0f) {
```

Comment:
Here is a different opinion that the version in the master is correct: https://stackoverflow.com/questions/56711084/represents-fp16-minimum-number-in-hex-format

This is saved in bits representation for which 0x0400 is correct.
Please elaborate why?

Reply:
Version in the master is not quite correct because it causes accuracy degradation :) `0x0004` is minimal positive **normalized** value but minimal positive **denormalized/subnormal** is `0x0001`, and by the link you provide this information is confirmed.

![image](https://user-images.githubusercontent.com/5703039/219029731-2e4961c4-6c1c-4f63-ba75-615ad6b1c148.png)

Therefore the current fix is correct